src/java/org/apache/cassandra/index/sai/disk/vector/VectorMemtableIndex.java

```diff
@@ -198,6 +198,13 @@ public RangeIterator search(QueryContext context, Expression expr, AbstractBound
         return new ReorderingRangeIterator(keyQueue.build(Comparator.naturalOrder()), keyQueue.size());
     }
 
+    @Override
+    public long estimateMatchingRowsCount(Expression expression, AbstractBounds<PartitionPosition> keyRange)
+    {
+        // TODO: this is a stub, we need to compute the correct fraction
+        return primaryKeys.size();
+    }
```

Comment:
Given `QueryController#estimateMatchingRowCount` is the only place we call this method and it doesn't handle ANN operations, do we even need a correct implementation here? It seems like the estimate only applies to search predicates, not order predicates, at the moment.

Reply:
Good catch. We don't. Because we don't use this estimation method for ANN at all.

Reply:
Now it throws UnsupportedOperationException